eclipsesource/theia
# [vscode] support TaskPresentationOptions close

#### What it does

_Adds the close property to TaskPresentationOptions_
This enables the possibility to close the terminal once the task is finished.

Fixes #12611

Contributed on behalf of ST Microelectronics

#### How to test 
1. Install following extension:
-  extension: [task-provider-samples-0.0.3.zip](https://github.com/eclipsesource/theia/files/11800018/task-provider-samples-0.0.3.zip)
- source: [task-provider-sample-0.0.3-src.zip](https://github.com/eclipsesource/theia/files/11800019/task-provider-sample-0.0.3-src.zip)

2. Copy this rakefile at the root of your workspace:  [Rakefile.zip](https://github.com/eclipsesource/theia/files/11800020/Rakefile.zip). It contains several tasks that can be identified by the sample extension. In the extension, the build tasks are supposed to close the terminal after the task is finished (`close` set to `true`) whereas the terminal should remain on completion for other tasks like test tasks.

3. On `master` theia, the terminal always remain after the tasks are finished (this is also the default behavior after the patch when `close` is not defined.). The extension is written so the tasks are configured differently if they are build tasks or any other kind of tasks. This does not do any difference on `master`, as the close property is not known.
![12611-before](https://github.com/eclipsesource/theia/assets/3964263/bd3fe826-7e50-4d02-87f5-9fd1e347ecd3)

4. Switch to this PR and run the same tests. The terminal will close for the build tasks, and remain open for reuse for the other tasks like test tasks. 
![12611-after](https://github.com/eclipsesource/theia/assets/3964263/25b6a4d6-50c8-49b9-b8ae-156bb079afa3)

#### Review checklist


#### Reminder for reviewers



* Adds close to the TaskPresentationOptions
* Checks when task is finished if terminal shall be closed

Contributed on behalf of STMicroelectronics



#### What it does


#### How to test


#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
